File: python/v2.3/get_floodlight_activity_groups.py
#!/usr/bin/python
#
# Copyright 2015 Google Inc. All Rights Reserved.
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#      http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.

"""This example displays floodlight activity groups for a given advertiser.

To create an advertiser, run create_advertiser.py.
"""

import argparse
import sys

import dfareporting_utils
from oauth2client import client

# Declare command-line flags.
argparser = argparse.ArgumentParser(add_help=False)
argparser.add_argument(
    'profile_id', type=int,
    help='The ID of the profile to list activity groups for')
argparser.add_argument(
    'advertiser_id', type=int,
    help='The ID of the advertiser to list activity groups for')


def main(argv):
  # Retrieve command line arguments.
  flags = dfareporting_utils.get_arguments(argv, __doc__, parents=[argparser])

  # Authenticate and construct service.
  service = dfareporting_utils.setup(flags)

  profile_id = flags.profile_id
  advertiser_id = flags.advertiser_id

  try:
    # Construct the request.
    request = service.floodlightActivityGroups().list(
        profileId=profile_id, advertiserId=advertiser_id)

    # Execute request and print response.
    response = request.execute()

    for group in response['floodlightActivityGroups']:
      print ('Found floodlight activity group with ID %s and name "%s".'
             % (group['id'], group['name']))

  except client.AccessTokenRefreshError:
    print ('The credentials have been revoked or expired, please re-run the '
           'application to re-authorize')


if __name__ == '__main__':
  main(sys.argv)

File: ruby/v3.2/floodlight/get_floodlight_activity_groups.rb
#!/usr/bin/env ruby

#
# Copyright:: Copyright 2016, Google Inc. All Rights Reserved.
#
# License:: Licensed under the Apache License, Version 2.0 (the "License");
#           you may not use this file except in compliance with the License.
#           You may obtain a copy of the License at
#
#           http://www.apache.org/licenses/LICENSE-2.0
#
#           Unless required by applicable law or agreed to in writing, software
#           distributed under the License is distributed on an "AS IS" BASIS,
#           WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
#           implied.
#           See the License for the specific language governing permissions and
#           limitations under the License.
#
# This example displays floodlight activity groups for a given advertiser.
#
# To create an advertiser, run create_advertiser.rb.

require_relative '../dfareporting_utils'

def get_floodlight_activity_groups(profile_id, advertiser_id)
  # Authenticate and initialize API service.
  service = DfareportingUtils.get_service

  token = nil
  loop do
    result = service.list_floodlight_activity_groups(profile_id,
      advertiser_id: advertiser_id,
      page_token: token,
      fields: 'nextPageToken,floodlightActivityGroups(id,name)')

    # Display results.
    if result.floodlight_activity_groups.any?
      result.floodlight_activity_groups.each do |group|
        puts format(
          'Found floodlight activity group with ID %d and name "%s".',
          group.id, group.name
        )
      end

      token = result.next_page_token
    else
      # Stop paging if there are no more results.
      token = nil
    end

    break if token.to_s.empty?
  end
end

if $PROGRAM_NAME == __FILE__
  # Retrieve command line arguments
  args = DfareportingUtils.get_arguments(ARGV, :profile_id, :advertiser_id)

  get_floodlight_activity_groups(args[:profile_id], args[:advertiser_id])
end
